### PR TITLE
Deleting messages from view and gauges on cwd editor

### PR DESCRIPTION
**Pull request by adartulloch**
This pull request is attempting to create an option to remove messages from "view" and "gauges". I do that by updating going to the DB and updating the message to an empty string and the probability to 0. However, there are a few bugs that I do not know how to fix. One is that when I call the onClick of the delete icon button to remove an entry from the div form, it visually does so, however it also gives me a console error when I try to switch to any other tab in the editor. I suspect is has to do with how I am handling the HTML in my deleteMessage function:
```
var elem = document.getElementById("row");
    if (elem.parentNode) {
      elem.parentNode.removeChild(elem);
    }
```
which should be properly deleting one item from the form.

### CODE REVIEW

**Review by Sammidysam — CHANGES_REQUESTED** [review]
We need a backend route for deleting a message.  Before this can be merged, we will need to move a pull request through the backend project adding a delete route.  Then, we also need to delete using the React way, as that is what is currently causing the console error.

Nice work though!  I can see you learning by doing.

File: src/MessageInput.js
Comment (adartulloch) [reply src/MessageInput.js:51]: I'm almost certain that the state should take care of it here...but I think I'll be able to tell if it doesn't.

**Comment by adartulloch** [comment]
I'm looking through the backend code now, and I'm assuming that I will have to write methods in `glyphs.js` for a route to delete?

File: src/MessageInput.js
Comment (Sammidysam) [reply src/MessageInput.js:51]: Certainly!  This is purely me just asking since I didn't bother checking myself.  Just something to keep in mind.

**Comment by Sammidysam** [comment]
Yep.  Feel free to contact @acrap5 if you would like a call intro to everything - he did say he can find a time on Slack.  Otherwise, you'll want to mimic messages where you have a function that does similar logic on both gauges and views.

**Comment by adartulloch** [comment]
Ok! I reached out to him and will proceed from here.
This is silly, but how do I create a pull request to a repository that I have commit access to? Do I still have to make a remote for the feature branch?

**Comment by Sammidysam** [comment]
You should be able to do this process:
- On `master`
- `git checkout -b new-feature`
- `git push -u origin new-feature`
- Go to GitHub and click "New pull request"

Does that answer the question fine?

**Comment by Sammidysam** [comment]
Hey, just wanted to check on this status of this because it has been a week?

**Comment by Sammidysam** [comment]
Sorry for not being clear.  In particular, I am interested in if work has been done on the backend to accommodate this frontend.

**Comment by adartulloch** [comment]
I moved to focus on CWD stuff before finishing the backend stuff up here as I thought Dashboard would be of a higher priority. What do you think?

**Comment by adartulloch** [comment]
But I went ahead with the backend in the mean time, and opened a PR there.

**Comment by Sammidysam** [comment]
Sounds great!  I got a chance to review that.  Could you also make the requested changes here that are irrelevant to the backend existing or not in the meantime?

**Comment by Sammidysam** [comment]
@adartulloch Just checking in on this since my last comments were a week ago.

**Comment by adartulloch** [comment]
Wow, I am so confused because this branch is missing both locally and on from `git branch -r`. I may open a new one...not quite sure what happened here. But I agree with the splicing, and I will get to it with the new backend routes that I believe should work

**Comment by adartulloch** [comment]
Oh, I see what happened. I forgot that I had originally forked this branch.

**Comment by Sammidysam** [comment]
So you're fine and all set to continue then?

**Comment by Sammidysam** [comment]
@adartulloch Do you have pending changes here or should I finish it up?

**Comment by adartulloch** [comment]
Didn't quite get a chance to finish up the React stuff!

**Comment by Sammidysam** [comment]
Closing this since we are delaying the backend route to the fall, and therefore this cannot work until the backend is working.  I hope Adar does not delete his fork in the meantime, because this branch looks basically done, but I would say using the `delete` method is still a bit weird just given that it's not used much.  Maybe the opinion there will change over time, though.

**Pull request closed without merging**
